### PR TITLE
Fixed suggest tangent vector build params in c#

### PR DESCRIPTION
**Pull request by natan-fernandes**

### CODE REVIEW

**Comment by natan-fernandes** [comment]
It was not possible to convert the function with the `out` keyword, so now is possible to have this function in C#:
```
bool suggestTangentVectorBuildParams(VertexElementSemantic targetSemantic,
            unsigned short& outSourceCoordSet, unsigned short& outIndex)
```

File: OgreMain/include/Ogre.i [hunk 1]
```diff
@@ -768,6 +769,16 @@ SHARED_PTR(Material);
 %ignore Ogre::Mesh::getBoneAssignmentIterator;
 %template(PoseList) std::vector<Ogre::Pose*>;
 %template(SubMeshList) std::vector<Ogre::SubMesh*>;
+%include "typemaps.i"
```
Comment (paroj) [inline OgreMain/include/Ogre.i:772]: this is already included further on top

File: OgreMain/include/Ogre.i [hunk 2]
```diff
@@ -768,6 +769,16 @@ SHARED_PTR(Material);
...
+%define %standard_byref_params(TYPE)
+
+       %apply TYPE& INOUT { TYPE& };
+
+       %apply TYPE& OUTPUT { TYPE& outSourceCoordSet };
+       %apply TYPE& OUTPUT { TYPE& outIndex };
+
+%enddef
+%standard_byref_params(unsigned short)
```
Comment (paroj) [inline OgreMain/include/Ogre.i:781]: please use a more specifc %apply like here:
https://github.com/OGRECave/ogre/blob/e84d43eb5ec4b2017d02c18621d6aa0b69dea67f/OgreMain/include/Ogre.i#L367

that macro hides what you are actually doing here.

**Review by paroj — CHANGES_REQUESTED** [review]

File: OgreMain/include/Ogre.i [hunk 3]
```diff
@@ -764,10 +765,15 @@ SHARED_PTR(Material);
...
-%ignore Ogre::Mesh::getPoseIterator;
+%ignore Ogre::Mesh::getPoseIterator;''
 %ignore Ogre::Mesh::getBoneAssignmentIterator;
 %template(PoseList) std::vector<Ogre::Pose*>;
 %template(SubMeshList) std::vector<Ogre::SubMesh*>;
+%define %standard_byref_params(unsigned short)
+       %apply Ogre::Mesh& INOUT { unsigned short& };
+       %apply Ogre::Mesh& OUTPUT { unsigned short& outSourceCoordSet, unsigned short& outIndex };
+%enddef
+%standard_byref_params(unsigned short)
```
Comment (paroj) [inline OgreMain/include/Ogre.i:776]: the whole standard_byref_params macro is superfluous

File: OgreMain/include/Ogre.i [hunk 4]
```diff
@@ -764,10 +765,15 @@ SHARED_PTR(Material);
...
 %ignore Ogre::Mesh::getPoseCount;
 %ignore Ogre::Mesh::getPose;
-%ignore Ogre::Mesh::getPoseIterator;
+%ignore Ogre::Mesh::getPoseIterator;''
 %ignore Ogre::Mesh::getBoneAssignmentIterator;
 %template(PoseList) std::vector<Ogre::Pose*>;
 %template(SubMeshList) std::vector<Ogre::SubMesh*>;
+%define %standard_byref_params(unsigned short)
+       %apply Ogre::Mesh& INOUT { unsigned short& };
+       %apply Ogre::Mesh& OUTPUT { unsigned short& outSourceCoordSet, unsigned short& outIndex };
```
Comment (paroj) [inline OgreMain/include/Ogre.i:774]:
```suggestion
       %apply unsigned short& OUTPUT { unsigned short& outSourceCoordSet, unsigned short& outIndex };
```

File: OgreMain/include/Ogre.i [hunk 5]
```diff
@@ -764,10 +765,15 @@ SHARED_PTR(Material);
 %ignore Ogre::Mesh::getSubMeshIterator;
 %ignore Ogre::Mesh::getPoseCount;
 %ignore Ogre::Mesh::getPose;
-%ignore Ogre::Mesh::getPoseIterator;
+%ignore Ogre::Mesh::getPoseIterator;''
 %ignore Ogre::Mesh::getBoneAssignmentIterator;
 %template(PoseList) std::vector<Ogre::Pose*>;
 %template(SubMeshList) std::vector<Ogre::SubMesh*>;
+%define %standard_byref_params(unsigned short)
+       %apply Ogre::Mesh& INOUT { unsigned short& };
```
Comment (paroj) [inline OgreMain/include/Ogre.i:773]:
```suggestion

```

File: OgreMain/include/Ogre.i [hunk 6]
```diff
@@ -768,6 +769,10 @@ SHARED_PTR(Material);
 %ignore Ogre::Mesh::getBoneAssignmentIterator;
 %template(PoseList) std::vector<Ogre::Pose*>;
 %template(SubMeshList) std::vector<Ogre::SubMesh*>;
+%define %standard_byref_params(unsigned short)
+       %apply unsigned short& OUTPUT { unsigned short& outSourceCoordSet, unsigned short& outIndex };
+%enddef
+%standard_byref_params(unsigned short)
```
Comment (paroj) [inline OgreMain/include/Ogre.i:775]:
```suggestion
%apply unsigned short& OUTPUT { unsigned short& outSourceCoordSet, unsigned short& outIndex };
```

File: OgreMain/include/Ogre.i [hunk 7]
```diff
@@ -768,6 +769,10 @@ SHARED_PTR(Material);
 %ignore Ogre::Mesh::getBoneAssignmentIterator;
 %template(PoseList) std::vector<Ogre::Pose*>;
 %template(SubMeshList) std::vector<Ogre::SubMesh*>;
+%define %standard_byref_params(unsigned short)
```
Comment (paroj) [inline OgreMain/include/Ogre.i:772]:
```suggestion
```

Comment (natan-fernandes) [reply OgreMain/include/Ogre.i]: We need to remove %enddef at the end of this block too I think

**Pull request merged**